tuned/profiles/functions/function_package2cpus.py

```diff
@@ -0,0 +1,63 @@
+import os
+import fnmatch
+
+from . import base
+from tuned.utils.commands import commands
+from tuned import logs
+
+log = logs.get()
+cmd = commands()
+
+SYSFS_DIR = "/sys/devices/system/cpu/"
+
+class package2cpus(base.Function):
+	"""
+	Provides cpu device list for a package (socket)
+	"""
+
+	def __init__(self):
+		super(package2cpus, self).__init__("package2cpus", 0)
+
+	def execute(self, args):
+		if not super(package2cpus, self).execute(args):
+			return None
+
+		if len(args) <= 0:
+			return None
+
+		try:
+			all_cpus = fnmatch.filter(os.listdir(SYSFS_DIR), "cpu[0-9]*")
+		except OSError:
+			return None
+
+		devices = []
+
+		for cpu in all_cpus:
+			f = SYSFS_DIR + cpu + "/topology/physical_package_id"
+			if not os.path.exists(f):
+				log.warning("File '%s' does not exist" % f)
+				continue
+
+			try:
+				value = cmd.read_file(f)
+				package_id = int(value)
+			except ValueError:
+				log.warning("Failed to get package id for cpu '%s'" % cpu)
```

Comment:
I think you could also log the content of the `value` which may help with debugging possible problems.